src/rules/noUnusedVariableRule.ts

```diff
@@ -32,13 +28,17 @@ class NoUnusedVariablesWalker extends Lint.RuleWalker {
     private skipBindingElement: boolean;
     private skipParameterDeclaration: boolean;
     private skipVariableDeclaration: boolean;
+
+    private fileName: string;
     private languageService: ts.LanguageService;
 
-    constructor(sourceFile: ts.SourceFile, options: Lint.IOptions, languageService: ts.LanguageService) {
+    constructor(sourceFile: ts.SourceFile, options: Lint.IOptions) {
         super(sourceFile, options);
         this.skipVariableDeclaration = false;
         this.skipParameterDeclaration = false;
-        this.languageService = languageService;
+
+        this.fileName = (Lint.isTsxFile(sourceFile.fileName) ? "file.tsx" : "file.ts");
+        this.languageService = Lint.createLanguageService(this.fileName, sourceFile.getFullText());
```

Comment:
hm, I liked it earlier when we passed a `LanguageService` to the constructor.... that makes the whole thing more testable... but I see why you're doing it this way so that you only have to declare `fileName` once.

why don't we just use `sourceFile.fileName`? why even make a new name at all?